### PR TITLE
LinkSessions, cleaning, java namespace change

### PR DESCRIPTION
**Pull request by Nightenom**
# Changes proposed in this pull request:
- LinkSessions (+ channels) and structure sharing
- Settings serializable into ByteBuf
- Cleanup and fixes everywhere
- changing namespace from `com.structurize.api` to `com.ldtteam.structurize.api` and from `com.structurize.coremod` to `com.ldtteam.structurize`
- reorganize (move) some directories (might need a build.gradle update ?)
- unfinished scan command

Review please

### CODE REVIEW

File: src/api/java/com/ldtteam/structurize/api/util/BlockUtils.java [hunk 1]
```diff
@@ -376,7 +376,7 @@ public static IBlockState getBlockStateFromStack(final ItemStack stack)
             return Blocks.LAVA.getDefaultState();
         }
 
-        return stack.getItem() instanceof ItemBlock ? ((ItemBlock) stack.getItem()).getBlock().getStateFromMeta(stack.getItemDamage()) : Blocks.GOLD_BLOCK.getDefaultState();
+        return stack.getItem() instanceof ItemBlock ? ((ItemBlock) stack.getItem()).getBlock().getDefaultState() : Blocks.GOLD_BLOCK.getDefaultState();
```
Comment (Raycoms) [inline src/api/java/com/ldtteam/structurize/api/util/BlockUtils.java]: it has this one yet

File: build.gradle [hunk 2]
```diff
@@ -136,10 +136,6 @@ artifacts {
     archives sourceJar
 }
 
-javadoc {
-    source += sourceSets.api.allSource
-}
```
Comment (Raycoms) [inline build.gradle:141]: no?

File: src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java [hunk 3]
```diff
@@ -31,7 +31,7 @@
     /**
      * This blocks name.
      */
-    private static final String BLOCK_NAME = "multiBlock";
+    private static final String BLOCK_NAME = "multiblock";
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java:34]: won't that break it?

Comment (Nightenom) [inline src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java]: @Raycoms this one, somewhere it's lowercased, somewhere with big B

File: build.gradle (hunk 2 above)
Comment (Nightenom) [reply build.gradle:141]: doubled

File: src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java (hunk 3 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java:34]: somewhere it's lowercased, somewhere with big B
probably yes

Comment (Nightenom) [reply src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java:34]: Wait I just dig deeper,
`setRegistryName(Constants.MOD_ID.toLowerCase() + ":" + BLOCK_NAME);` was using big B
`GameRegistry.registerTileEntity(TileEntityMultiBlock.class, new ResourceLocation(Constants.MOD_ID, MultiBlock.getName()));` was using small b
So what was the actual registered name?

Comment (marchermans) [reply src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java:34]: SetRegistryname auto lowercases,
It should always be lowercase!

Comment (Nightenom) [reply src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java:34]: yeah, I agree, but
setRegistryName only lowercases namespace, not the path
translationKey doesn't care
registerTileEntity use the same method for checking as setRegistryName (so only lowercases namespace)

Comment (Nightenom) [reply src/main/java/com/ldtteam/structurize/blocks/MultiBlock.java:34]:
```java
int index = name.lastIndexOf(':');
String oldPrefix = index == -1 ? "" : name.substring(0, index).toLowerCase(Locale.ROOT);
name = index == -1 ? name : name.substring(index + 1);
```

**Comment by marchermans** [comment]
Why closing?

**Comment by Nightenom** [comment]
Due to weird and many merge conflicts

**Pull request closed without merging**